lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -36,6 +36,52 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $allColumns = $this->_conn->fetchAll($this->_platform->getListAllColumnsSQL($currentDatabase));
+
+        // Get all foreign keys definitions in one database call.
+        $allForeignKeys = $this->_conn->fetchAll($this->_platform->getListAllForeignKeysSQL($currentDatabase));
+
+        // Get all indexes definitions in one database call.
+        $allIndexes = $this->_conn->fetchAll($this->_platform->getListAllIndexesSQL($currentDatabase));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = rtrim(ltrim($tableName, '"'), '"');
+
+            // Process columns for this table.
+            $tableColumns = array_filter($allColumns, function ($column) use ($unquotedTableName) {
```

Comment:
The list of columns is iterated/filtered once for every single table which is suboptimal. Instead, it could be re-grouped by table name once, and then every table could pick its own group if it exists.